### DISCUSSION TITLE
Why did you use transactions in queries?

### DISCUSSION THREAD

**Discussion by alexxty7** [benbjohnson/wtf#20] [Q&A] [answered]
Hi, 
I Want to ask you What the reason to wrap all queries with a transaction. 
As I know, we should use transactions if we need the ability to rollback every change. 
There is any reason to use it?

**Comment by benbjohnson** [comment, marked as answer]
I find it easier to compose lower level helper functions if they all use a `*sql.Tx`. If there was a performance hit and I needed to optimize then I would consider removing them then.
